pkg/cmd/get/vault/config/get_vault_config.go

```diff
@@ -115,7 +116,11 @@ func (o *GetVaultConfigOptions) systemVaultClient() (vault.Client, error) {
 
 	vaultConfig, err := vault.FromMap(installValues, devNamespace)
 	if err != nil {
-		return nil, errors.Wrapf(err, "unable to parse Vault configuration from ConfigMap %s", kube.ConfigMapNameJXInstallConfig)
+		if !vaultConfig.IsExternal() {
```

Comment:
actually this is an issue we already had in _CreateSystemVaultClient_. I did not realize  this could become an issue here as well. Still would love to hear how this actually becomes an issue. 

The rub is that it is the wrong thing to do to create the Vault config instance at this point. In `factory.go` I do:

```
	if installValues[vault.URL] != "" {
		vaultConfig, err := vault.FromMap(installValues, namespace)
		if err != nil {
			return nil, errors.Wrapf(err, "error parsing Vault configuration from ConfigMap %s in namespace %s", kube.ConfigMapNameJXInstallConfig, namespace)
		}
		return f.CreateExternalVaultClient(vaultConfig, kubeClient)
	}

	name, err := f.getVaultName(namespace)
	if err != nil {
		return nil, err
	}
	return f.CreateInternalVaultClient(name, namespace)
```

Not ideal, since now I have the means on how to check whether a config is external or internal at multiple places. The best option would probably be to refactor the `Vault` code a bit and separate creation of the instance from its validation. 

Also in this particular case  I am wondering why the code is not calling `CreateSystemVaultClient` directly. That's what it is after, right?

Reply:
See issue https://github.com/jenkins-x/jx/issues/7315

Reply:
I didn't think it was necessary to validate the new configuration when calling `jx get vault-config`?

Reply:
> Also in this particular case I am wondering why the code is not calling `CreateSystemVaultClient` directly. That's what it is after, right?

I believe so. Would you expect to use `jx get vault-config` for an external vault setup?

Reply:
>  Would you expect to use jx get vault-config for an external vault setup?

absolutely. For the Jenkins X system vault that is. It does not matter in this case whether this Vault instance is external (installed and managed independently of Jenkins X) or internal (created by Jenkins X using the Bankk-Vaults Operator). 

You need to be able to connect to your system Vault to rotate secrets for example.

Really, the best is to call CreateSystemVaultClient from _factory.go_. In fact I seem to recall that I intended to do this at some stage, but forgot about it.

Reply:
Will take a look at this one now 👍